crates/matrix-sdk/src/widget/machine/tests/send_event.rs

```diff
@@ -0,0 +1,40 @@
+use std::time::Duration;
+
+use ruma::{api::client::future::FutureParameters, events::TimelineEventType};
+
+use super::WIDGET_ID;
+use crate::widget::machine::{
+    from_widget::FromWidgetRequest,
+    incoming::{IncomingWidgetMessage, IncomingWidgetMessageKind},
+};
+
+#[test]
+fn parse_future_action() {
+    let raw = json_string!({
+        "api": "fromWidget",
+        "widgetId": WIDGET_ID,
+        "requestId": "send_event-request-id",
+        "action": "send_event",
+        "data": {
+            "content": {},
+            "future_timeout": 10000,
+            "room_id": "!rXAYvblqYaGiJmeRdR:matrix.org",
+            "state_key": "_@abc:example.org_VFKPEKYWMP",
+            "type": "org.matrix.msc3401.call.member",
+        },
+    });
+    if let IncomingWidgetMessageKind::Request(a) =
```

Comment:
I think you want `assert_let` here to make sure we fail if this is the wrong type for some reason.